app/controllers/articles.server.controller.js

```diff
@@ -88,9 +88,20 @@ exports.list = function(req, res) {
  * Article middleware
  */
 exports.articleByID = function(req, res, next, id) {
+
+	if (!mongoose.Types.ObjectId.isValid(id)) {
+		return res.status(400).send({
+			message: 'Article is invalid'
+		});
+	}
+
 	Article.findById(id).populate('user', 'displayName').exec(function(err, article) {
 		if (err) return next(err);
-		if (!article) return next(new Error('Failed to load article ' + id));
+		if (!article) {
+			res.status(404).send({
```

Comment:
shouldn't we `return res.status.send` so that next() doesn't get called below? Or we could toss it into an `else {}`?